src/substrate/subscribeFunc.ts

```diff
@@ -104,24 +103,37 @@ export const subscribeEvents: SubscribeFunc<ApiPromise, Block, ISubscribeOptions
         && (!offlineRange || !offlineRange.startBlock || offlineRange.startBlock < lastBlockNumber)) {
       offlineRange = { startBlock: lastBlockNumber };
     }
+    // if running in archival mode and can't determine the offlineRange then start from Block 0
+    if(archival && (!offlineRange || !offlineRange.startBlock)){
+      offlineRange = {startBlock: 0}
+    }
 
     // if we can't figure out when the last block we saw was, do nothing
     // (i.e. don't try and fetch all events from block 0 onward)
-    if (!offlineRange || !offlineRange.startBlock) {
-      log.warn('Unable to determine offline time range.');
-      return;
-    }
-
-    // poll the missed blocks for events
-    try {
-      const blocks = await poller.poll(offlineRange);
-      await Promise.all(blocks.map(processBlockFn));
-    } catch (e) {
-      log.error(`Block polling failed after disconnect at block ${offlineRange.startBlock}`);
+    // if (!offlineRange || !offlineRange.startBlock) {
+    //   log.warn('Unable to determine offline time range.');
+    //   return;
+    // }
+
+    // if running in archival mode then fetch blocks from the chain in batches 
+    // from starting block to the head of the node
+    if(archival){
```

Comment:
This should really go either in an external script, or in the poller itself, with something like `poller.archive` that does the entire loop. It doesn't make sense to have it in the "top level" `subscribeFunc`.

Reply:
moved the logic to `poller.archive` method.